lang/python/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+# See the file LICENSE for redistribution information.
+#
+
+project(python_swig C)
+
+# UseSWIG: Policy option to use the standard target name over a generated name.
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 NEW)
+endif()
+
+# UseSWIG: Policy option to honor SWIG_MODULE_NAME.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+
+set(python_libs)
+set(python_version)
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.12.0")
+    # This method of finding python libs has been deprecated since version 3.12.
+    # If we are running with a greater CMake version, opt to use the Python3 package.
+    set(Python_ADDITIONAL_VERSIONS 3.9 3.8 3.7 3.6 3.5)
+    find_package(PythonInterp REQUIRED)
+    find_package(PythonLibs REQUIRED)
+    include_directories(${PYTHON_INCLUDE_DIRS})
+    set(python_libs ${PYTHON_LIBRARIES})
+    set(python_version ${PYTHON_VERSION_STRING})
+else()
+    find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+    include_directories(${Python3_INCLUDE_DIRS})
+    set(python_libs ${Python3_LIBRARIES})
+    set(python_version ${Python3_VERSION})
+endif()
+
+# Manually specify the interface name '_wiredtiger' over a generated name (for backwards compatibility).
+list(APPEND swig_flags "-interface;_wiredtiger")
+
+list(APPEND swig_flags "-DPY_MAJOR_VERSION=${VERSION_MAJOR}")
```

Comment:
Can you help pointing to me where `PY_MAJOR_VERSION` symbol gets used?

Reply:
We don't have any specific usages of `PY_MAJOR_VERSION` in our code though the symbol tells CPython at compile time whether the C code its compiling with will be used with Python 2 or 3. SWIG also generates checks with `PY_MAJOR_VERSION` in the python C wrapper code to tell whether it can use Python3 compatible macros. 
I was a bit lazy here using `${VERSION_MAJOR}` to satisfy the `PY_MAJOR_VERSION >= 3` condition and forgot to update it (thanks for pointing this out!) so I made it more explicit to just lock it at `3`. 

It's not super necessary to define and can remove it if you want, its probably defined in somewhere in the user installed python headers. I just figured it might be good practice to be explicit on supporting only the Python3 API & ABI.

Reply:
Thanks for the detailed explanation. I'm fine to keep it with explicit support for Python3.